docs/Extensions.md

```diff
@@ -8,6 +8,11 @@ Packages are the main building blocks for applications and will most often consi
 ## Plugins
 Plugins are often smaller packages that are NOT to be used directly in a project but in conjunction with a package to add some additional functionality.
 
+## Abstract Extensions
+Abstract extensions are something that is not meant to be used directly inside a user project but rather used to enhance and define something that another extensions can build on. They are from a technical standpoint identical to their non abstract counterparts. The only difference is the naming convention where `roc-abstract-*` is used over either `roc-package-*` or `roc-plugin-*`. It's up to the developer to select a fitting name for an abstract package that fits the naming convention. For a abstract package that might be `roc-abstract-package-*` and for an abstract plugin `roc-abstract-plugin-*`.
```

Comment:
Abstract extensions are not meant to be used directly inside a user project but rather to enhance and define something that other extensions can build on. They are from a technical standpoint identical to their non abstract counterparts. The only difference is the naming convention where `roc-abstract-*` is used over either `roc-package-*` or `roc-plugin-*`. It's up to the developer to select a fitting name for an abstract package that fits the naming convention. For an abstract package that might be `roc-abstract-package-*` and for an abstract plugin `roc-abstract-plugin-*`.